system/util/dummy_diag_publisher/src/dummy_diag_publisher_node/dummy_diag_publisher_node.cpp

```diff
@@ -63,26 +79,46 @@ void DummyDiagPublisherNode::produceDiagnostics(diagnostic_updater::DiagnosticSt
   stat.summary(status.level, status.message);
 }
 
-void DummyDiagPublisherNode::onTimer(const ros::TimerEvent & event)
+void DummyDiagPublisherNode::onTimer()
 {
   if (config_.is_active) {
     updater_.force_update();
   }
 }
 
 DummyDiagPublisherNode::DummyDiagPublisherNode()
+: Node("dummy_diag_publisher"),
+  update_rate_(declare_parameter("update_rate", 10.0)),
+  updater_(this)
 {
-  // Parameter
-  private_nh_.param("update_rate", update_rate_, 10.0);
-  diag_config_ = getConfig<DiagConfig>(private_nh_, "diag_config");
+  // Get configuration
+  std::map<std::string, std::string> configuration_parameters;
+  configuration_parameters.insert(std::pair<std::string, std::string>("diag_config.name", ""));
+  configuration_parameters.insert(std::pair<std::string, std::string>("diag_config.hardware_id", ""));
+  configuration_parameters.insert(std::pair<std::string, std::string>("diag_config.msg_ok", ""));
+  configuration_parameters.insert(std::pair<std::string, std::string>("diag_config.msg_warn", ""));
+  configuration_parameters.insert(std::pair<std::string, std::string>("diag_config.msg_error", ""));
+  configuration_parameters.insert(std::pair<std::string, std::string>("diag_config.msg_stale", ""));
 
-  // Dynamic Reconfigure
-  dynamic_reconfigure_.setCallback(boost::bind(&DummyDiagPublisherNode::onConfig, this, _1, _2));
+  this->declare_parameters<std::string>("", configuration_parameters);
+  this->get_parameters("diag_config", configuration_parameters);
+  diag_config_ = DiagConfig(configuration_parameters);
+  
+  // set parameter callback
```

Comment:
We have need to declare parameter in order to set it from parameter callback.

```suggestion
  // set parameter callback
  config_.status = static_cast<Status>(this->declare_parameter("status", 0));
  config_.is_active = this->declare_parameter("is_active", false);
```